programs/drift/src/math/amm_spread.rs

```diff
@@ -153,6 +155,37 @@ pub fn calculate_long_short_vol_spread(
     ))
 }
 
+pub fn calculate_inventory_liquidity_ratio(
+    base_asset_amount_with_amm: i128,
+    base_asset_reserve: u128,
+    min_base_asset_reserve: u128,
+    max_base_asset_reserve: u128,
+) -> DriftResult<i128> {
+    // computes min(1, x/(1-x)) for 0 < x < 1
+
+    // inventory scale
+    let (max_bids, max_asks) = _calculate_market_open_bids_asks(
+        base_asset_reserve,
+        min_base_asset_reserve,
+        max_base_asset_reserve,
+    )?;
+
+    let min_side_liquidity = max_bids.min(max_asks.abs());
+    msg!("max_bids={:?} / max_asks={:?}", max_bids, max_asks);
+
+    // cap so (6e9 * AMM_RESERVE_PRECISION)^2 < 2^127
+    let amm_inventory_pct = if base_asset_amount_with_amm < 6000000000000000000 {
+        base_asset_amount_with_amm
+            .abs()
+            .safe_mul(PERCENTAGE_PRECISION_I128)?
+            .safe_div(min_side_liquidity.max(1))?
```

Comment:
could maybe just max this with `PERCENTAGE_PRECISION_I128` and get rid of the min before, might make it easier to replicate in ts sdk?